### PR TITLE
feat: add AWS Backup recovery browser

### LINKED ISSUE #321: feat: add AWS Backup recovery browser
## Summary

Add an AWS Backup browser for vaults, protected resources, recovery points, and recent backup-job failures.

## Details

- List backup vaults with lock/encryption metadata and recovery-point counts.
- Drill into recovery points with resource type/name, creation and completion times, status, lifecycle dates, size, and source vault.
- Show protected resources and recent failed or expired backup jobs as recovery-readiness signals.
- Sort failed jobs and expiring/failed recovery points first while keeping pagination deterministic.
- Keep partial results visible when an individual vault or recovery-point lookup is denied.
- Initial scope is read-only; backup, restore, delete, and vault-lock mutations are excluded.
- AWS API reference: https://docs.aws.amazon.com/aws-backup/latest/APIReference/API_ListRecoveryPointsByBackupVault.html

## Checklist

- [ ] Add AWS Backup service and feature identifiers in `internal/domain/model.go` and `internal/domain/catalog.go`
- [ ] Add vault, recovery-point, protected-resource, and backup-job models under `internal/services/aws/`
- [ ] Add a mockable Backup client and paginated repository methods in `internal/services/aws/repository.go`
- [ ] Add vault list and scrollable recovery-point/resource/job detail screens under `internal/app/`
- [ ] Cover empty accounts, pagination, partial authorization failures, lifecycle dates, and stable sorting
- [ ] Add mock-client tests following existing service patterns
- [ ] Update `README.md` and `docs/service-specific-highlights.md`
- [ ] Run `make test` and `make build`

### PR DESCRIPTION
**Pull request by YoungJinJung**
### Summary

- add paginated AWS Backup vault discovery and a read-only recovery posture detail view
- show recovery points, lifecycle timing, protected resources, recent failed jobs, vault lock state, and per-section partial-result warnings
- integrate filtering, saved views, context/region resets, help, tests, the AWS SDK dependency, and user-facing documentation

README's duplicated global-filter paragraph is intentionally unchanged because #318 owns that exact cleanup. After #318 lands, this branch should be rebased while retaining AWS Backup in the single remaining inventory.

### Related Issues

Closes #321

### Validation

- make test
- make build
- go vet ./...
- go test -race ./internal/services/aws ./internal/app -run Backup -count=1
- git diff --check

### Checklist

- [x] Scope is focused
- [x] Branch name follows docs/branch-naming-harness.md
- [x] Documentation harness reviewed (docs/documentation-harness.md)
- [x] README updated if user-facing behavior changed
- [x] Relevant docs/ pages updated if architecture, auth, config, or workflow changed
- [x] Tests/validation included
- [x] Breaking changes documented (none)

## Summary by CodeRabbit

* **New Features**
  * Added a read-only AWS Backup browser for backup vaults.
  * View recovery points, protected resources, encryption and Vault Lock details, and recent failed or expired jobs.
  * Added filtering, keyboard navigation, scrolling, refresh, and drill-down views.
  * Partial results remain available with inline warnings when some data cannot be loaded.
* **Documentation**
  * Updated project, architecture, and feature documentation to describe AWS Backup support.

### CODE REVIEW

**Review by YoungJinJung — COMMENTED** [review]
Reviewed head: db8c2581a0c01674b4ef592b28c89fd1ca5dadc6

## Findings

1. **P2 — One-sided Vault Lock bounds render as a false `N-0 days` range** (`internal/services/aws/backup.go:186-187`, `internal/app/screen_backup.go:438-440`)

   `MinRetentionDays` and `MaxRetentionDays` are optional AWS fields, but mapping them through `aws.ToInt64` discards whether each bound was absent. `backupLockSummary` then formats both values whenever either is non-zero. A common lock with only a 7-day minimum therefore appears as `locked (7-0 days)`, incorrectly telling operators that the maximum is zero days.

   Preserve presence for each retention bound (pointers or explicit known flags) and render the minimum and maximum independently; add coverage for minimum-only, maximum-only, and unbounded lock configurations.

2. **P2 — Missing recovery-point sizes are displayed as real zero-byte backups** (`internal/services/aws/backup.go:207`, `internal/app/screen_backup.go:379`)

   `BackupSizeInBytes` is optional, but `aws.ToInt64(nil)` collapses “not reported” to `0`, and the detail view renders that as `0 B`. For resource types or responses where AWS omits the field, the recovery browser therefore presents unknown size as an actual empty backup, which can mislead recovery-readiness triage.

   Preserve whether the SDK field was present and render `-`/`unknown` when absent while retaining `0 B` for an explicitly reported zero; add a mapping/rendering test that distinguishes nil from zero.

**Comment by YoungJinJung** [comment]
Applied the YoungJinJung review findings in commit 9312bb2. Preserved Vault Lock retention-bound presence for minimum-only, maximum-only, and unbounded policies; preserved recovery-point size presence so omitted values render as a dash while explicit zero remains 0 B; and added mapping/rendering regression coverage. Local validation passed: targeted Backup tests, make test, make build, and git diff --check.

File: internal/app/screen_backup.go [hunk 1]
```diff
@@ -0,0 +1,452 @@
...
+		if msg.err != nil {
+			bm.finishError(m, msg.err)
+			return *m, nil, true
+		}
+		bm.detail = msg.detail
+		bm.detailErrors = msg.warnings
+		bm.detailScroll = 0
+		bm.errorActive = false
+		finishBackupLoad(m, screenBackupVaultDetail)
+		return *m, nil, true
```
[the remark this answers is not in the document]
Comment (YoungJinJung) [reply internal/app/screen_backup.go:116]: Verified in b760dc4: Backup list and detail commands already run through the shared startLoadingFor lifecycle, which binds results to a generation and drops stale genBoundMsg values before submodels receive them. A Backup-local counter would duplicate that guard. Added a regression covering an old-context detail result for a same-named vault.

File: internal/services/aws/backup_model.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	return strings.ToLower(strings.Join([]string{v.ARN, v.Name, v.Region, v.State, v.Type, v.EncryptionKeyARN, v.EncryptionKeyType}, " "))
+}
+
+// DisplayTitle returns a column-aligned vault row.
+func (v BackupVault) DisplayTitle() string {
+	lock := "-"
+	if v.Locked {
+		lock = "locked"
+	}
+	return fmt.Sprintf("%-32.32s  %-12.12s  %8d  %-8s  %s", v.Name, valueOrDash(v.State), v.RecoveryPointCount, lock, valueOrDash(v.Type))
```
[the remark this answers is not in the document]
Comment (YoungJinJung) [reply internal/services/aws/backup_model.go]: Fixed in b760dc4 by moving vault-row layout into the TUI layer with Lipgloss display-width columns. Added a Unicode vault-name regression that checks row and header alignment.

File: internal/services/aws/backup.go [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+}
+
+func sortBackupJobs(items []BackupJob) {
+	sort.SliceStable(items, func(i, j int) bool {
+		left, right := backupJobStatusRank(items[i].State), backupJobStatusRank(items[j].State)
+		if left != right {
+			return left < right
+		}
+		return items[i].CreatedAt.After(items[j].CreatedAt)
+	})
```
[the remark this answers is not in the document]
Comment (YoungJinJung) [reply internal/services/aws/backup.go:185]: Fixed in b760dc4: protected resources and recovery points now use normalized ARN as the final key, and jobs use normalized ID. Added equal-primary-key regression coverage for all three sorts.

**Comment by YoungJinJung** [comment]
Addressed the current review findings in b760dc4. Detail scrolling now uses one warning-aware window calculation; unfiltered saved views clear prior feature filters; Backup vault rows use Lipgloss display-width columns; detail sorts have unique final keys; and stale same-name prior-context results are covered through the existing shared generation guard. Validation passed: targeted tests, make test, make build, go vet ./..., focused race tests, and git diff --check. The issue checklist reference to docs/service-specific-highlights.md cannot be updated because that file does not exist on current main; the feature coverage is already documented in README.md and both project overview and architecture documents.

**Review by YoungJinJung — COMMENTED** [review]
Reviewed head: b760dc436299f9a22dd1f9e8ddb7c3741f9e5387

## Findings

1. **P2 — Detail refresh leaves vault-level recovery posture stale** (`internal/app/screen_backup.go:232-235`, `internal/services/aws/backup.go:56-57`)

   Pressing `r` says it is refreshing the selected vault, but `loadDetail` passes the original list-row snapshot into `GetBackupVaultDetail`, which copies that `BackupVault` and only reloads recovery points, protected resources, and jobs. Vault state, recovery-point count, encryption metadata, and especially Vault Lock/retention settings can therefore remain stale indefinitely even while the refreshed screen looks current, which can mislead recovery-readiness checks.

   Re-fetch the selected vault metadata as part of the detail refresh (for example, reuse the existing vault-list call and match by ARN/name, or add `DescribeBackupVault` and document its permission), then add a refresh regression that changes lock/count metadata between the initial list and refreshed detail.

**Review by YoungJinJung — COMMENTED** [review]
Reviewed current head b760dc436299f9a22dd1f9e8ddb7c3741f9e5387. The optional metadata fixes, warning-aware scrolling, saved-view filter reset, Unicode column layout, deterministic tie-breakers, and shared generation guard all hold in the complete diff; all current inline threads are resolved and I found no new actionable code issue. Local make test, make build, focused Backup race tests, and git diff --check pass. Do not merge yet: #318 must land first, then rebase and add AWS Backup to the single retained filtering inventory; the current head also still needs an Amazon Q result and an independent approval satisfying branch protection.

**Comment by YoungJinJung** [comment]
Applied the current-head vault-refresh finding in cfc31b7. Explicit detail refreshes now re-list and re-identify the selected vault before loading recovery points, protected resources, and jobs, so recovery-point count, encryption, and Vault Lock metadata are current without adding a new AWS permission. The refreshed vault also replaces the selected detail snapshot, and a missing vault fails instead of presenting stale posture. Added a regression that changes recovery-point count and Vault Lock retention between the initial list and refresh. Validation passed: focused Backup tests, make test, make build, focused Backup race tests, and git diff --check.

**Review by YoungJinJung — COMMENTED** [review]
Reviewed head: cfc31b710ec1d222e7fd65cd39682574b8094900

## Findings

1. **P2 — A detail refresh is discarded after returning to the vault list** (`internal/app/screen_backup.go:104-108`)

   The detail completion replaces only `selected` and `detail`; `vaults` and `filtered` still contain the pre-refresh `BackupVault`. After `r` observes a changed recovery-point count or Vault Lock policy, pressing Esc therefore renders the old row, and pressing Enter again selects that stale row and calls `loadDetail(..., false)`, copying the old vault metadata back into the detail screen. Reconcile `msg.detail.Vault` into the canonical vault slice and reapply the active filter/index before setting the selection (the existing DynamoDB `reconcileTable` flow is a useful pattern). Add a regression covering refresh → Esc → Enter so refreshed metadata remains current.

2. **P2 — An incomplete paginated refresh is reported as vault deletion** (`internal/app/screen_backup.go:274-281`)

   `ListBackupVaults` intentionally returns completed pages plus `listWarnings` and a nil fatal error when a later page fails. If the selected vault was on that unavailable later page, `findBackupVault` returns false and this branch raises “is no longer available,” discards the pagination warning, and routes the user to the fatal error screen even though the refresh never established that the vault was deleted. When the list is partial, preserve the prior vault metadata and surface the warning (or otherwise report the pagination failure without claiming deletion); reserve the unavailable error for a complete list. Add a refresh test where an earlier page succeeds and the selected vault’s page fails.

**Review by YoungJinJung — COMMENTED** [review]
Reviewed head: cfc31b710ec1d222e7fd65cd39682574b8094900

## Finding

1. **P2 — A partial refresh can falsely report the selected vault as deleted** (`internal/app/screen_backup.go:273-282`)

   `ListBackupVaults` deliberately returns completed pages with `listWarnings` when a later page fails. If the selected vault was on that failed page, `findBackupVault` cannot find it and this path returns `backup vault … is no longer available`, replacing the usable detail with a fatal error. That contradicts the PR’s partial-result behavior and turns a transient pagination denial into false deletion posture.

   Only conclude that the vault disappeared when the re-list completed without warnings. When the list is partial and the vault is absent, retain the existing vault snapshot, continue loading its detail, and surface the list warning. Add a refresh regression with one completed page followed by an error where the selected vault is not in the retained page.

**Comment by YoungJinJung** [comment]
Applied the current-head review findings in b05acf4. Refreshed vault metadata is now reconciled into the canonical and filtered list state, so refresh → back → reopen keeps the new recovery count and Vault Lock posture. If a later vault-list page fails and the selected vault is absent from the retained pages, refresh now preserves the prior vault snapshot, continues loading detail, and surfaces the pagination warning instead of claiming deletion. Added regressions for both flows. Validation passed: focused Backup refresh tests, focused race tests, make test, make build, and git diff --check.

**Review by YoungJinJung — COMMENTED** [review]
Reviewed head: b05acf497fb4aed561375ba967d1c503fc6157ae

## Findings

1. **P2 — Subsequent detail failures are hidden while affected sections render as empty** (`internal/app/screen_backup.go:403-405`)

   The repository deliberately returns one warning for each independently loaded section, but this view passes only `bm.detailErrors[0]` to `renderWarningSummary`. When two sections fail—or when a refresh pagination warning precedes a section failure—the remaining failures are not identified anywhere, while those sections still render `None available` or `None in the AWS Backup 30-day job window`. That makes unavailable recovery-posture data look like a confirmed empty result.

   Render every detail warning, or retain per-section availability and render an explicit unavailable state in the affected section. Make `detailVisibleLines` account for the resulting warning height, and add a view test with two distinct section errors that verifies both are visible and the final detail lines remain reachable.

**Review by YoungJinJung — COMMENTED** [review]
Reviewed head: b05acf497fb4aed561375ba967d1c503fc6157ae

## Finding

1. **P2 — Preserve why a completed job needs attention** (internal/services/aws/backup.go:142-145, 250-259; internal/app/screen_backup.go:544-557)

   backupJobNeedsAttention includes a COMPLETED job when MessageCategory is non-empty and not SUCCESS, but mapBackupJob drops MessageCategory and the detail view renders only the state. If AWS omits StatusMessage, the job therefore appears under “Recent Failed / Expired Jobs” simply as COMPLETED, so the operator cannot tell why an apparently successful job was flagged; this also does not substantiate the README promise to show completed-with-issues jobs.

   Carry MessageCategory (or a derived completed-with-issues display status) into BackupJob and render it, then add a regression for COMPLETED plus a non-SUCCESS category that verifies the issue is visible while a SUCCESS job remains excluded.

Validation on this head: make test, make build, focused Backup race tests, and git diff --check all pass.

**Comment by YoungJinJung** [comment]
Applied both current-head review findings in b9d6c63. The detail view now renders every lookup warning and subtracts each rendered warning line from the scroll window. Backup jobs now retain and render MessageCategory, so completed-with-issues jobs explain why they need attention while SUCCESS jobs remain excluded. Validation passed: focused Backup tests, make test, make build, focused Backup race tests, and git diff --check.

**Comment by YoungJinJung** [comment]
Addressed the valid Amazon Q cancellation finding in 575f15a: detail loading now stops between independent AWS sections when the context is canceled, with a regression proving later section calls are skipped. The reconcileVault shadowing claim is a false positive: the inner index intentionally selects the matching row in bm.filtered after the canonical bm.vaults update, and the outer index is no longer used. Validation passed again: focused Backup tests, make test, make build, focused Backup race tests, and git diff --check.

**Review by YoungJinJung — COMMENTED** [review]
Reviewed current head `575f15acf7a57286bcc9cac856e1400458a4b332`. The multi-warning rendering, completed-with-issues job context, and between-section cancellation fixes resolve the latest findings; I read the complete diff, tests, CI, prior reviews, inline threads, and new commits and found no new actionable code issue. Local `make test`, `make build`, `go vet ./...`, `go test -race ./internal/services/aws ./internal/app -run Backup -count=1`, and `git diff --check origin/main...HEAD` pass. Do not merge yet: #318 must land first, then rebase this branch and add AWS Backup only to the single retained filtering inventory; an independent approval satisfying branch protection is still required.

**Review by YoungJinJung — COMMENTED** [review]
Reviewed head: 575f15acf7a57286bcc9cac856e1400458a4b332

## Findings

1. **P2 — Preserve Backup load completions while Add Context is open** (`internal/app/screen_backup.go:57-70`)

   `finishBackupLoad` relies on `backupOverlayPrevious` to replace the `screenLoading` return target when a result arrives behind an overlay, but this switch does not handle `screenContextAdd`. A user can start a Backup load, press `C`, enter Add Context with `a`, and receive the Backup result while that screen is active. The handler consumes the result but leaves `ctxPrevScreen == screenLoading`; after exiting Add Context and then the context picker, normal back-navigation lands on an orphaned loading screen with no command left to complete it.

   Include `screenContextAdd` in the Backup overlay chain (following `ctxPrevScreen`, as the context-picker flow ultimately does), or preserve that pending context return explicitly. Add a regression that delivers a vault or error completion while `screenContextAdd` is active and verifies that exiting the context flow returns to the completed Backup target instead of `screenLoading`.

**Comment by YoungJinJung** [comment]
Applied the current-head Add Context finding in `7c27e38`. Backup success and error completions now traverse `screenContextAdd` through the existing context return chain, so closing Add Context and the context picker returns to the completed Backup target instead of an orphaned loading screen. Added an end-to-end regression for vault completion through that exit flow. Validation passed: focused Backup navigation tests, `make test`, `make build`, `gofmt -d`, and `git diff --check`.

**Review by YoungJinJung — COMMENTED** [review]
Re-reviewed current head `7c27e38e484e44f0c86a9906870085989816c7ce`. The Add Context overlay traversal now preserves Backup success/error completions behind `screenContextAdd`, and the regression covers returning through the context picker to the completed vault list. I read the complete diff, tests, CI, prior reviews, inline threads, and the new commit and found no new actionable code issue. Local `make test`, `make build`, `go vet ./...`, `go test -race ./internal/services/aws ./internal/app -run Backup -count=1`, and `git diff --check origin/main...HEAD` pass. I did not approve or merge: #318 still owns the prerequisite README deduplication/rebase, and branch protection requires an independent approval that this PR author identity cannot provide.

**Review by youngjinjung-linq — COMMENTED** [review]
Independent review at head 7c27e38e484e44f0c86a9906870085989816c7ce: no new actionable code findings. Full test suite, go vet, build, Backup-focused race tests, diff, CI, and prior threads were checked. Approval is deferred because #318 must land first and this branch must reconcile the single README filtering inventory; please request re-review on the rebased head.
